src/DIRAC/FrameworkSystem/scripts/dirac_login.py

```diff
@@ -1,237 +1,380 @@
+        :param _arg: unuse
+        """
+        os.environ["DIRAC_USE_ACCESS_TOKEN"] = "false"
+        self.result = "proxy"
+        return S_OK()
 
-        :return: S_OK()
+    def setToken(self, _arg) -> S_OK:
+        """Return tokens
+
+        :param _arg: unuse
         """
-        self.lifetime = arg
+        os.environ["DIRAC_USE_ACCESS_TOKEN"] = "true"
+        self.useDIRACAS(None)
+        self.result = "token"
         return S_OK()
 
+    def authStatus(self, _arg) -> S_OK:
+        """Get authorization status
+
+        :param _arg: unuse
+        """
+        result = self.getAuthStatus()
+        if result["OK"]:
+            self.howToSwitch()
+            DIRAC.exit(0)
+        gLogger.fatal(result["Message"])
+        DIRAC.exit(1)
+
     def registerCLISwitches(self):
         """Register CLI switches"""
-        Script.registerSwitch("", "info", "output current user authorization status", self.getInfo)
-        Script.registerSwitch("P", "proxy", "request a proxy certificate with DIRAC group extension", self.returnProxy)
-        Script.registerSwitch("g:", "group=", "set DIRAC group", self.setGroup)
-        Script.registerSwitch("I:", "issuer=", "set issuer", self.setIssuer)
-        Script.registerSwitch("T:", "lifetime=", "set proxy lifetime in a hours", self.setLivetime)
-        Script.registerSwitch("F:", "file=", "set token file location", self.setTokenFile)
+        Script.registerArgument(
+            "group: select a DIRAC group for authorization, can be determined later.", mandatory=False
+        )
+        Script.registerArgument(["scope: scope to add to authorization request."], mandatory=False)
+        Script.registerSwitch("T:", "lifetime=", "set access lifetime in a hours", self.setLivetime)
+        Script.registerSwitch(
+            "O:",
+            "save-output=",
+            "where to save the authorization result(e.g: proxy or tokens). By default we will try to find a standard place.",
+            self.setOutputFile,
+        )
+        Script.registerSwitch("I:", "issuer=", "set issuer.", self.setIssuer)
+        Script.registerSwitch(
+            " ",
+            "use-certificate",
+            "in case you want to generate a proxy using a certificate. By default.",
+            self.useCertificate,
+        )
+        Script.registerSwitch(
+            " ", "use-diracas", "in case you want to authorize with DIRAC Authorization Server.", self.useDIRACAS
+        )
+        Script.registerSwitch("C:", "certificate=", "user certificate location", self.setCertificate)
+        Script.registerSwitch("K:", "key=", "user key location", self.setPrivateKey)
+        Script.registerSwitch(" ", "proxy", "return proxy in case of successful authorization", self.setProxy)
+        Script.registerSwitch(" ", "token", "return tokens in case of successful authorization", self.setToken)
+        Script.registerSwitch(" ", "status", "print user authorization status", self.authStatus)
```

Comment:
```suggestion
        Script.registerSwitch("", "proxy", "return proxy in case of successful authorization", self.setProxy)
        Script.registerSwitch("", "token", "return tokens in case of successful authorization", self.setToken)
        Script.registerSwitch("", "status", "print user authorization status", self.authStatus)
```